### ISSUE TITLE
When using SURROGATE_OR_BUSINESS_KEY in the business key for equals JPA_GETTER should be enforced.

### ISSUE THREAD

**Issue by Moritz-Hoeppner** [jqno/equalsverifier#892]
**Describe the bug**
I had a bug in my project where equals used `.id` instead of `.getId()` when comparing the IDs of an Object. The `.id` just returned `null` while `.getId` returned the correct value. I believe that for surrogate key equals, the getters should be enforced. 

**To Reproduce**

**Code that triggers the behavior**
Equals Method:
```java
 @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof MyEntity that)) {
            return false;
        }
        return id != null && id.equals(that.id);
    }
```
The test should make sure to use 
`return id != null && id.equals(that.**getId()**); instead`

Test that passed:
```java
@Test
    void testEquals() {
        EqualsVerifier
            .forClass(MyEntity.class)
            .withPrefabValues(Set.class, Set.of(new MyEntity()), Set.of(new MyEntity()))
            .suppress(Warning.SURROGATE_KEY)
            .suppress(Warning.IDENTICAL_COPY_FOR_VERSIONED_ENTITY)
            .suppress(Warning.STRICT_HASHCODE)
            .verify();
    }
```

_Remember, you have the problem right in front of you. I have to try and reproduce it in my spare time. Please help me, so that I can help you!_

**Error message**
The test is passing so no error message. Also no warning.

**Expected behavior**
I would expect the test to fail. But it did not.

**Version**
3.15.4

**Additional context**
According to this blog post, the equals should also be implemented using `getId()` : https://thorben-janssen.com/ultimate-guide-to-implementing-equals-and-hashcode-with-hibernate/

**Comment by jqno** [comment]
Am I correct in assuming you're referring to the "Using a Generated Primary Key" section of that blog post, where the id is generated by Hibernate and is only available once the entity is persisted?

Can you describe the circumstances when entity's `id` field is null, but `getId()` isn't? When exactly does that happen?

**Comment by MoritzOtto** [comment]
Hey jqno,
we do have the Problem, if the Entity is "lazy" loaded and a HibernateProxyElement got created. The easiest way to create this is calling the getReferenceById() Method by Spring Boot.
When doing this JPA acts the same way then reference a relationship described here: https://jqno.nl/equalsverifier/errormessages/jpa-direct-reference-instead-of-getter/
Hope this helps already...

**Comment by jqno** [comment]
Thanks for the update, and sorry for the slow replies - it's been a busy week.

I've released version 3.15.5, and updated the documentation page you reference as well.
